docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1740,201 @@ As a response to blocked request AdGuard returns a short video placeholder.
+* [`$other`](#other-modifier),
+* [`$ping`](#ping-modifier),
+* [`$script`](#script-modifier),
+* [`$stylesheet`](#stylesheet-modifier),
+* [`$subdocument`](#subdocument-modifier),
+* [`$websocket`](#websocket-modifier),
+* [`$xmlhttprequest`](#xmlhttprequest-modifier);
+
+This also includes rules that implicitly add the modifier `$document`:
+* [`$popup`](#popup-modifier);
+
+or special exceptions that implicitly add `$document,subdocument`:
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
+
+The presence of any content-type modifiers adds `(50 + 50 / N)`, where `N` is the number of modifiers present, for example:
+`||example.com^$image,script` will add `50 + 50 / 2 = 50 + 25 = 75` to the total weight of the rule. The `$popup` also belongs to this category, because it implicitly adds the modifier `$document`. Similarly, specific exceptions add `$document,subdocument`.
+
+#### 3. $domain or $app with allowed domains or applications: {#priority-category-3}
+
+Specified domains through `$domain` or specified applications through `$app` add `100 + 100 / N`, where `N` is the number of modifier values for example:
+`||example.com^$domain=example.com|example.org|example.net` will add `100 + 100 / 3 = 134.3 = 135` or
+`||example.com^$app=org.example.app1|org.example.app2` will add `100 + 100 / 2 = 151` or
+`||example.com^$domain=example.com,app=org.example.app1|org.example.app2` will add `100 + 100/1` ($domain part) and `100 + 100/2` ($app part) - will add `350` in total.
+
+#### 4. Specific exceptions: {#priority-category-4}
+
+[//]: # (Please keep them sorted)
+
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
+
+Each of which adds `10^3` to the priority.
+
+As well as exception with [`$document modifier`](#document-modifier): because it's an alias for `$elemhide,content,jsinject,urlblock,extension`.
+It will add `10^3` for each modifier from [the top list](#priority-category-4), `10^3 * 5` in total.
+
+In addition, each of these exceptions implicitly adds the two allowed content-type modifiers `$document,subdocument`.
+
+#### 5. Allowlist rule {#priority-category-5}
+
+Modifier `@@` adds `10^4` to rule priority.
+
+#### 6. $redirect rules {#priority-category-6}
+
+[//]: # (Please keep them sorted)
+
+* [`$redirect`](#redirect-modifier),
+* [`$redirect-rule`](#redirect-rule-modifier).
+
+Each of which adds `10^5` to rule priority.
+
+#### 7. important rules {#priority-category-7}
```

Comment:
We currently have tests such as these:
```
        { // document > important
                .rules = { "@@||test1", "@@||test2$important", "@@||test3$document" },
                .expected_rule = "@@||test3$document",
        },
```
```
        { // exceptional document > important blocking document
                .rules = { "@@||test1$document", "||test2$document,important" },
                .expected_rule = "@@||test1$document",
        },
```
So, document exceptions seem to have special priority boost that is not reflected in this spec. Is this an intended change, or is the spec incorrect?

Reply:
@ngorskikh according to the documentation `$important` has a higher priority than any exceptions